test/e2e/validation_test.go

```diff
@@ -143,3 +150,261 @@ func testNodeTaint(t *testing.T) {
+	roleClient := tc.client.K8s.RbacV1().Roles(tc.workloadNamespace)
+	err := roleClient.Delete(context.TODO(), tc.workloadNamespace, meta.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return errors.Wrap(err, "unable to delete existing role")
+	}
+	role := rbac.Role{
+		TypeMeta:   meta.TypeMeta{},
+		ObjectMeta: meta.ObjectMeta{Name: tc.workloadNamespace},
+		Rules: []rbac.PolicyRule{
+			{
+				Verbs:         []string{"use"},
+				APIGroups:     []string{"security.openshift.io"},
+				Resources:     []string{"securitycontextconstraints"},
+				ResourceNames: []string{"hostnetwork"},
+			},
+		},
+	}
+	_, err = roleClient.Create(context.TODO(), &role, meta.CreateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "unable to create role")
+	}
+	return nil
+}
+
+// ensureTestRunnerRoleBinding ensures the proper RoleBinding exists, a requirement for SSHing into a Windows node
+func (tc *testContext) ensureTestRunnerRoleBinding() error {
+	rbClient := tc.client.K8s.RbacV1().RoleBindings(tc.workloadNamespace)
+	err := rbClient.Delete(context.TODO(), tc.workloadNamespace, meta.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return errors.Wrap(err, "unable to delete existing role")
+	}
+	rb := rbac.RoleBinding{
+		TypeMeta:   meta.TypeMeta{},
+		ObjectMeta: meta.ObjectMeta{Name: tc.workloadNamespace},
+		Subjects: []rbac.Subject{{
+			Kind:      "ServiceAccount",
+			APIGroup:  "",
+			Name:      tc.workloadNamespace,
+			Namespace: tc.workloadNamespace,
+		}},
+		RoleRef: rbac.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     tc.workloadNamespace,
+		},
+	}
+	_, err = rbClient.Create(context.TODO(), &rb, meta.CreateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "unable to create role")
+	}
+	return nil
+}
+
+// sshSetup creates all the Kubernetes resources required to SSH into a Windows node
+func (tc *testContext) sshSetup() error {
+	if err := tc.ensureTestRunnerSA(); err != nil {
+		return errors.Wrap(err, "error ensuring SA created")
+	}
+	if err := tc.ensureTestRunnerRole(); err != nil {
+		return errors.Wrap(err, "error ensuring Role created")
+	}
+	if err := tc.ensureTestRunnerRoleBinding(); err != nil {
+		return errors.Wrap(err, "error ensuring RoleBinding created")
+	}
+	return nil
+}
+
+// copyKeySecret copies the current private-key secret from the WMCO namespace into the testContext's test namespace
+func (tc *testContext) copyKeySecret() error {
+	currentSecret, err := tc.client.K8s.CoreV1().Secrets(tc.namespace).Get(context.TODO(), secrets.PrivateKeySecret, meta.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "unable to get current private key secret from WMCO namespace")
+	}
+
+	// Remove secret from test namespace if it already exists, so the new secret can be created
+	secretClient := tc.client.K8s.CoreV1().Secrets(tc.workloadNamespace)
+	err = secretClient.Delete(context.TODO(), secrets.PrivateKeySecret, meta.DeleteOptions{})
```

Comment:
I again don't follow why you have to delete it?

Reply:
+1 
Should this noop if the key already exists in the test namespace, similar to `sshSetup`?

Reply:
This has not been addressed.

Reply:
See https://github.com/openshift/windows-machine-config-operator/pull/474#discussion_r657425994